src/main/java/com/joejoe2/chat/service/channel/PrivateChannelService.java

```diff
@@ -61,8 +61,21 @@ SliceList<PrivateChannelProfile> getAllChannels(String ofUserId, PageRequest pag
    * @return profile of target channel
    * @throws UserDoesNotExist
    * @throws ChannelDoesNotExist
-   * @throws InvalidOperation target user is not in members of target channel
+   * @throws InvalidOperation user is not in members of target channel
    */
   PrivateChannelProfile getChannelProfile(String ofUserId, String channelId)
       throws UserDoesNotExist, ChannelDoesNotExist, InvalidOperation;
+
+  /**
+   * let user block or unblock the channel(another user)
+   *
+   * @param userId user id of the initiator
+   * @param channelId id of target channel
+   * @param isBlock block or unblock
+   * @throws UserDoesNotExist
+   * @throws ChannelDoesNotExist
+   * @throws InvalidOperation user is not in members of target channel
+   */
+  void block(String userId, String channelId, boolean isBlock)
```

Comment:
I think - blockUser sound the best here

Reply:
I thought user block the given channel(implicit refer to another member), so I write block here.

Reply:
Thanks for the clarification. Sounds reasonable.